katdal/sensordata.py

```diff
@@ -126,34 +186,112 @@ class TelstateSensorData(SensorData):
     Parameters
     ----------
     data : recarray-like, with fields ('timestamp', 'value')
-        Uncached sensor data as record array or equivalent (such as a
-        :class:`h5py.Dataset`)
+        Uninterpolated sensor data as structured array or equivalent (such as
+        an :class:`h5py.Dataset`)
```

Comment:
All SensorData contents will be wrapped. It's only once it hits CategoricalData that it is unwrapped.

The big picture is this:

telstate / HDF5: raw input format

SensorData: just a wrapper per (unextracted) sensor to give a common API to various input formats
CategoricalData / ndarray: replaces SensorData after extraction, contains actual data
SensorCache: dict-like collection of the above (both extracted and unextracted sensors mixed together)